### PR TITLE
BUILD-3061 Migrate CI from Azure to Cirrus

### PR DESCRIPTION
**Pull request by sebastienvermeille**
# BUILD-3061 Migrate CI from Azure to Cirrus

## Changes
* Migrate azure pipeline to cirrus
* Adapt README.md links

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,75 +1,17 @@
...
-          fields: repo,author,eventName
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_BUILD_WEBHOOK }}
+  release:
+    permissions:
+      id-token: write
+      contents: write
+    uses: SonarSource/gh-action_release/.github/workflows/main.yaml@v5
+    with:
+      publishToBinaries: true
```
Comment (julien-carsique-sonarsource) [inline .github/workflows/release.yml:15]:
```suggestion
      publishToBinaries: false
```

File: .cirrus.yml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    - build
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+    cpu: 2
+    memory: 2G
+    # TODO: check with sonarlint team if they accept that, was not the case before
+    # run only on master and long-term branches
+  only_if: $CIRRUS_USER_COLLABORATOR == 'true' && ($CIRRUS_BRANCH == "master" || $CIRRUS_BRANCH =~ "branch-.*")
+  env:
+    WS_APIKEY: VAULT[development/kv/data/mend data.apikey]
```
Comment (julien-carsique-sonarsource) [inline .cirrus.yml:110]: Also set `WS_PRODUCTNAME: SonarLint/Core`

**Review by julien-carsique-sonarsource — APPROVED** [review]

File: .cirrus.yml [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+      path: "whitesource/**/*"
+
+qa_task:
+  # Run ITs
+  depends_on:
+    - build
+    - test_windows
+  env:
+    ARTIFACTORY_API_KEY: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-private-reader access_token]
+    GITHUB_TOKEN: VAULT[development/github/token/licenses-ro token]
```
Comment (julien-carsique-sonarsource) [inline .cirrus.yml:149]: OOMKilled, try this:
```suggestion
    GITHUB_TOKEN: VAULT[development/github/token/licenses-ro token]
    MAVEN_OPTS: '-Xmx5632m'
```
If it does not work or slows down the build, then you can try to increase the memory a bit more.

**Review by julien-carsique-sonarsource — APPROVED** [review]

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻

**Review by julien-carsique-sonarsource — APPROVED** [review]

**Pull request merged**